File: packages/OWASP/2025/top-10/supply-chain-vulnerabilities/zig/test_security.zig
const std = @import("std"); const lib = @import("lib.zig"); const vulnerable = @import("vulnerable.zig");
test "vulnerability" { try std.testing.expect(std.mem.indexOf(u8, vulnerable.vulnerable_scenario("x http://"), "http://") != null); }
test "defense" { const cleaned = try lib.LLM_1ntruder.clean(std.testing.allocator, "Supply Chain Vulnerabilities", "x http://"); try std.testing.expect(std.mem.indexOf(u8, cleaned, "http://") == null); }

File: packages/OWASP/2025/top-10/supply-chain-vulnerabilities/py/test_security.py
import unittest
from vulnerable import vulnerable_scenario
from lib import LLM_1ntruder
class SecurityTest(unittest.TestCase):
 def test_vulnerability(self): self.assertIn("http://", vulnerable_scenario("x http://"))
 def test_defense(self): self.assertNotIn("http://", LLM_1ntruder.clean("Supply Chain Vulnerabilities", "x http://"))
if __name__ == "__main__": unittest.main()
